target/linux/realtek/files-5.10/drivers/net/phy/rtl83xx-phy.c

```diff
@@ -2952,13 +3107,594 @@ int rtl9300_sds_cmu_band_get(int sds)
 	return cmu_band;
 }
 
-int rtl9300_configure_serdes(struct phy_device *phydev)
+static int rtl8218d_serdes_mode_get(struct phy_device *phydev)
+{
+	u32 block;
+       u32 data;
+	int mode;
+
+	block = phy_read_paged(phydev, RTL8XXX_PAGE_MAIN, 30);
+	phy_write_paged(phydev, RTL8XXX_PAGE_MAIN, 30, 8);
+	data = phy_read_paged(phydev, 0x260, 18);
+	phy_write_paged(phydev, RTL8XXX_PAGE_MAIN, 30, block);
+	pr_info("%s, reading phy %d got me block %04x and data %04x\n",
+		__func__, phydev->mdio.addr, block, data);
+
+	switch (data & 0xf0) {
+	case 0xd0:
+		mode = PHY_INTERFACE_MODE_QSGMII;
+		break;
+	case 0xb0:
+		mode = PHY_INTERFACE_MODE_XGMII;
+		break;
+	default:
+		pr_err("%s: unknown SDS mode: %x\n", __func__, data & 0xf0);
+		return PHY_INTERFACE_MODE_NA;
+	}
+
+	pr_info("%s: SDS mode: %x\n", __func__, data & 0xf0);
+	return mode;
+}
+
+static u32 rtl8218d_serdes_model_get(struct phy_device *phydev)
+{
+	u32 block
```

Comment:
semicolon missing

Reply:
yeah, sorry I pushed an intermediate state into this MR that I shouldn't have. Fixed now though.